cmdstanpy/stanfit.py

```diff
@@ -939,10 +923,9 @@ def diagnose(self) -> Optional[str]:
         """
         cmd_path = os.path.join(cmdstan_path(), 'bin', 'diagnose' + EXTENSION)
         cmd = [cmd_path] + self.runset.csv_files
-        result = do_command(cmd=cmd)
-        if result:
-            get_logger().info(result)
-        return result
+        result = StringIO()
+        do_command(cmd=cmd, show_console=True, fd_out=result)
+        return result.getvalue()
```

Comment:
I guess this always was the behavior, but printing out and returning seems strange to me. In repls/jupyter this leads to duplicated output

Reply:
agreed.  changing the return type would make sense - we could change it to bool for problems/no problems.
changing return type of a method seems like bad behavoir, but that's why we're still in beta.

Reply:
I think if I had to pick one behavior over the other I'd pick returning the string, and removing the side effect

Reply:
Yes, I think returning string is a better idea?